src/meta/Meta.js

```diff
@@ -0,0 +1,115 @@
+     *
+     * @param options See CaskSDK "meta" configuration.
+     */
+    constructor(options={}) {
+        this.options = options;
+        this.metaProvider = providers.NONE;
+        this.gasLimitMargin = 1;
+
+        if (options.biconomyApiKey) {
+            this.metaProvider = providers.BICONOMY;
+            this.gasLimitMargin = 1.15; // increase gasLimit by 15% to cover relay operations
+        }
+
+        this.logger = new Logger('CaskSDK::Meta', options.logLevel);
+    }
+
+    async init(ethersConnection) {
+        this.logger.info(`Initializing Cask Meta service.`);
+        if (!ethersConnection) {
+            this.logger.info("Cannot perform meta transactions without a connection");
+            return;
+        }
+        if (!ethersConnection.signer) {
+            this.logger.info("Cannot perform meta transactions without signer");
+            return;
+        }
+
+        switch (this.metaProvider) {
+            case providers.NONE:
+                break;
+            case providers.BICONOMY:
+                const chainInfo = chains.lookupChain(ethersConnection.chainId);
+                if (!chainInfo?.biconomyEnabled) {
+                    this.logger.info(`Biconomy meta transactions not supported on current chain ${ethersConnection.chainId}.`);
+                    break;
+                }
+                if (!this.options.biconomyApiKey[ethersConnection.chainId]) {
+                    this.logger.info(`Biconomy API Key not provided for current chain ${ethersConnection.chainId}.`);
+                    break;
+                }
+                if (ethersConnection.signer?.provider?.provider?.isBiconomy) {
+                    // already setup
+                    this.logger.info(`Biconomy already initalized.`);
+                    return;
+                }
+
+                try {
+                    this.biconomy = new Biconomy(ethersConnection.signer.provider.provider, {
+                        walletProvider: ethersConnection.signer.provider.provider,
+                        apiKey: this.options.biconomyApiKey[ethersConnection.chainId],
+                        debug: this.options.logLevel === 'debug',
+                    });
+                    this.logger.info(`Biconomy instantiated.`);
+
+                    ethersConnection.onEstimateGas(async (request, gas) => {
+                        if (gas && gas.toNumber() && this.gasLimitMargin) {
+                            let gasLimit = gas?.toNumber();
+                            let newGas = gasLimit * this.gasLimitMargin;
+                            return ethers.BigNumber.from(parseInt(newGas));
+                        }
+                        return gas;
+                    });
+                    this.logger.info(`Biconomy estimateGas patched.`);
+                } catch (error) {
+                    this.logger.info(`Biconomy failed to initialize.`);
+                    console.dir(error);
```

Comment:
@smokeynotes I've been keeping `console` messages out of the SDK. Can you just this.logger.warn that instead?

Reply:
Will do.